config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/PolicyEntityBuilder.java

```diff
@@ -260,16 +273,48 @@ private Encass getEncass(Bundle bundle, String name) {
         LOGGER.log(Level.FINE, "Looking for referenced encass: {0}", name);
         final AtomicReference<Encass> referenceEncass = new AtomicReference<>(bundle.getEncasses().get(name));
         if (referenceEncass.get() == null) {
+            Set<BundleDefinedEntities> bundleMetadataSet = bundle.getMetadataDependencyBundles();
+            Encass encass = null;
+            //check the dependency in the metadata file
+            if (bundleMetadataSet != null && !bundleMetadataSet.isEmpty()) {
+                for (BundleDefinedEntities bundleMetadata : bundleMetadataSet) {
+                    Collection<BundleDefinedEntities.DefaultMetadata> metadataCollection = bundleMetadata.getDefinedEntities();
+                    if (metadataCollection != null) {
+                        Optional<BundleDefinedEntities.DefaultMetadata> optionalMetadata = metadataCollection.stream().
+                                filter(metadata -> metadata.getName().equals(name)
+                                        && metadata.getType().equals(EntityTypes.ENCAPSULATED_ASSERTION_TYPE)).findFirst();
+                        if (optionalMetadata.isPresent()) {
+                            encass = getEncassFromMetadata(optionalMetadata.get());
+                            encass.setName(name);
+                        }
+                    }
+                    if (encass != null && !referenceEncass.compareAndSet(null, encass)) {
+                        throw new EntityBuilderException("Found multiple encasses in dependency bundles with name: " + name);
+                    }
+                }
+            }
+
+            //check the dependency in the given dependent bundle
             bundle.getDependencies().forEach(b -> {
-                Encass encass = b.getEncasses().get(name);
-                if (encass != null && !referenceEncass.compareAndSet(null, encass)) {
+                Encass encassDependency = b.getEncasses().get(name);
+                if (encassDependency != null && !referenceEncass.compareAndSet(null, encassDependency)) {
                     throw new EntityBuilderException("Found multiple encasses in dependency bundles with name: " + name);
                 }
             });
         }
         return referenceEncass.get();
     }
 
+    private Encass getEncassFromMetadata(final Metadata metadata) {
+        Encass encass = new Encass();
+        encass.setId(metadata.getId());
+        encass.setGuid(metadata.getGuid());
+        Set<Annotation> annotations = new HashSet<>();
+        annotations.add(AnnotableEntity.REUSABLE_ANNOTATION);
```

Comment:
are we adding the @reusable flag to policy if it is found in another bundle file?